packages/xod-project/src/flatten.js

```diff
@@ -85,7 +85,7 @@ const terminalOriginalDirectionLens = R.lensProp('originalDirection');
 const isLeafPatchWithImplsOrTerminal = def(
   'isLeafPatchWithImplsOrTerminal :: Patch -> Boolean',
   R.anyPass([
-    Patch.hasImpl,
+    R.both(Patch.isPatchNotImplementedInXod, Patch.hasImpl),
```

Comment:
This check looks strange: if an apple is green and greenish.

Why can’t we have a single function to check for the implementation presence? That is, we should kill either `Patch.isPatchNotImplementedInXod` or `Patch.hasImpl`